### DISCUSSION TITLE
how to split the bot code into separate files?

### DISCUSSION THREAD

**Discussion by Vyakoo** [PrismarineJS/mineflayer#3117] [Q&A]
I want to split the bot code into different files. For example, the code of walking, fishing, protection, etc. and at the end I import them in the main js file and call them, but how do I pass our bot object. I can't create a new bot because it won't be him anymore.

**Comment by PondWader** [comment]
You can export a function in the files and pass the bot.

```js
// otherModule.js
/**
 * @param {import('mineflayer').Bot} bot
*/
module.exports = (bot) => {
  // do stuff
}
```

```js
// index.js
const bot = createBot({ /* ... */ })

// Load modules
require('./otherModule.js')(bot)
```
